src/formatting-model-api.d.ts

```diff
@@ -0,0 +1,582 @@
+/**
+ * Formatting Pane Model Interfaces
+ */
+
+declare module powerbi {
+    export namespace visuals {
+        export type FormattingModelWarnings = { [cardUid: string]: IVisualWarning };
+
+        export interface FormattingModel {
+            cards: (FormattingCard | FormattingCardPlaceholder)[];
+            modelUnavailableImageName?: string;
+            modelUnavailableMessageKey?: string;
+            modelUnavailableLearnMoreLink?: string;
+            // this is used for displaying messages in formatting pane, with an action to usually open some dialog.
+            formattingModelMessage?: FormattingModelMessage;
+            warnings?: FormattingModelWarnings;
+        }
+
+        export interface FormattingModelMessage {
+            messageKey: string;
+            action?: FormattingModelMessageAction;
+        }
+
+        export interface FormattingModelMessageAction {
+            key: string;
+            fn: () => void;
+        }
+
+        export interface FormattingWarning {
+            warningMessage?: IVisualErrorMessage;
+        }
+
+        export interface FormattingCard extends VisualFormattingUid, VisualFormattingDisplayName, FormattingWarning {
+            disabled?: boolean;
+            /** card disabled reason */
+            disabledReason?: string;
+            /** This is top-level card On/Off property */
+            topLevelToggle?: EnabledSlice;
+            groups: (FormattingGroup | FormattingGroupPlaceholder)[];
+            /** This contains the list of descriptors for revert to default, note the descriptors of all the slices ever possible in the card must be added */
+            revertToDefaultDescriptors?: FormattingDescriptor[];
+            /** if true, this card should be populated into the analytics pane */
+            analyticsPane?: boolean;
+        }
+
+        export interface FormattingGroup extends VisualFormattingUid, VisualFormattingDisplayName {
+            disabled?: boolean;
+            /** card disabled reason */
+            disabledReason?: string;
+            /** This is top-level card On/Off property */
+            topLevelToggle?: EnabledSlice;
+            slices?: (FormattingSlice | FormattingSlicePlaceholder)[];
+            container?: FormattingContainer;
+            containerDisabled?: boolean;
+            /** Allows slices in the same group as containers. Slices are outside of the container's context */
+            sliceWithContainer?: boolean;
+            /**
+             * If delaySaveSlices is true, then this group's slices' value changes won't be saved to the visual until a
+             * signal action is taken. E.g., for an Analytics Pane forecast, the forecast parameter values shouldn't be
+             * saved to the visual until the Apply button is clicked. Note that this applies to all slices in the group.
+             */
+            delaySaveSlices?: boolean;
```

Comment:
is this applicable to CVs?

Reply:
I dont know exactly what is the use of this